docs/remote/create-dev-container.md

```diff
@@ -56,253 +43,162 @@ All of the predefined container configurations you can pick from come from the [
-sudo apt-get update
-sudo apt-get install <package>
-```
+![Select Dockerfile](images/containers/select-dockerfile.png)
 
-However, if you **rebuild** the container, you will have to **reinstall** anything you've installed manually. To avoid this problem, you can either use a series of commands in the `postCreateCommand` property in `devcontainer.json` or the `RUN` instruction [in a custom Dockerfile](#using-an-image-or-dockerfile). You can use `&&` to string together multiple commands.
+Now that you have a `devcontainer.json` and Dockerfile, let's see the general process for editing container configuration files.
 
-Using a Dockerfile:
+### Full configuration edit loop
 
-```Dockerfile
-RUN apt-get update && apt-get install <packaging>
-```
+Editing your container configuration is easy. Since rebuilding a container will "reset" the container to its starting contents (with the exception of your local source code), VS Code does not automatically rebuild if you edit a container configuration file (`devcontainer.json`, `Dockerfile`, and `docker-compose.yml`, which we'll get to next). Instead, there are several commands that can be used to make editing your configuration easier.
 
-Using `devcontainer.json`:
+Here is the typical edit loop using these commands:
 
-```json
-"postCreateCommand": "apt-get update && apt-get install <package>"
-```
+![Container edit loop illustration](images/containers/container-edit-loop.png)
 
-Or if running as a [non-root user](/docs/remote/containers-advanced.md#adding-a-nonroot-user-to-your-dev-container):
+1. Start with **Remote-Containers: Add Development Container Configuration Files...** in the Command Palette (`kbstyle(F1)`).
+2. Edit the contents of the `.devcontainer` folder as required.
+3. Try it with **Remote-Containers: Reopen Folder in Container**.
+4. If you see an error, click on **Open Folder Locally** in the dialog that appears.
+5. After the window reloads, a copy of the **build log will appear** in the console so you can investigate the problem. Edit the contents of the `.devcontainer` folder as required. (You can also use the **Remote-Containers: Open Log File...** command to see the log again if you close it.)
+6. Run **Remote-Containers: Rebuild and Reopen Folder in Container** and jump to step 4 if needed.
 
-```json
-"postCreateCommand": "sudo apt-get update && sudo apt-get install <package>"
-```
+If you already have a successful build, you can still edit the contents of the `.devcontainer` folder as required when connected to the container and then select **Remote-Containers: Rebuild Container** in the Command Palette (`kbstyle(F1)`) so the changes take effect.
 
-The `postCreateCommand` is run once the container is running, so you can also use the property to run commands like `npm install` or to execute a shell script in your source tree (if you have mounted it).
+You can also iterate on your container when using the **Remote-Containers: Open Repository in Container** command.
```

Comment:
FYI - Getting rid of this difference is one of the things I'm hoping gets done in the Sept iteration - this is where I'd mentioned Erich and Christof in that other doc. Adds complexity where we really don't need / want it.

Reply:
So the flow for working with repositories in containers would match the 6 steps above more closely rather than having its own unique 6 steps here? Do you think we should modify/remove anything for this doc as it currently stands, or wait until changes potentially happen in the September iteration?